crates/matrix-sdk/src/test_utils/mocks.rs

```diff
@@ -1697,23 +1697,70 @@ impl<'a> MockEndpoint<'a, RoomMessagesEndpoint> {
     ///
     /// Note: pass `chunk` in the correct order: topological for forward
     /// pagination, reverse topological for backwards pagination.
-    pub fn ok(
-        self,
-        start: String,
-        end: Option<String>,
-        chunk: Vec<impl Into<Raw<AnyTimelineEvent>>>,
-        state: Vec<Raw<AnyStateEvent>>,
-    ) -> MatrixMock<'a> {
-        let mock = self.mock.respond_with(ResponseTemplate::new(200).set_body_json(json!({
-            "start": start,
-            "end": end,
-            "chunk": chunk.into_iter().map(|ev| ev.into()).collect::<Vec<_>>(),
-            "state": state,
-        })));
+    pub fn ok(self, response: RoomMessagesResponse) -> MatrixMock<'a> {
+        let mut template = ResponseTemplate::new(200).set_body_json(json!({
+            "start": response.start,
+            "end": response.end,
+            "chunk": response.chunk,
+            "state": response.state,
+        }));
+
+        if let Some(delay) = response.delay {
+            template = template.set_delay(delay);
+        }
+
+        let mock = self.mock.respond_with(template);
         MatrixMock { server: self.server, mock }
     }
 }
 
+/// A response to a [`RoomMessagesEndpoint`] query.
+pub struct RoomMessagesResponse {
```

Comment:
Can we call this a bit differently? It's not the real Ruma response we use everywhere else, it's a test helper. Perhaps `RoomMessageResponseTemplate`?

Reply:
`MockRoomMessages`?

Reply:
Well we're not mocking, and it's not a message. It's about the response that you would send out if you were to send a room message. Then we pass that response to the mock server which will mock things for us.

I picked the `Template` naming because wiremock, which we use and wrap here, is using that term as well: https://docs.rs/wiremock/latest/wiremock/struct.ResponseTemplate.html#.